pandas/io/formats/format.py

```diff
@@ -565,315 +583,446 @@ def __init__(
+            str_columns = self._get_formatted_column_labels(frame)
 
-            if self.show_row_idx_names:
-                for x in str_columns:
-                    x.append("")
+        if self.show_row_idx_names:
+            for x in str_columns:
+                x.append("")
 
-            stringified = []
-            for i, c in enumerate(frame):
-                cheader = str_columns[i]
-                header_colwidth = max(
-                    col_space.get(c, 0), *(self.adj.len(x) for x in cheader)
-                )
-                fmt_values = self._format_col(i)
-                fmt_values = _make_fixed_width(
-                    fmt_values, self.justify, minimum=header_colwidth, adj=self.adj
-                )
+        strcols = []
+        for i, c in enumerate(frame):
+            cheader = str_columns[i]
+            header_colwidth = max(
+                col_space.get(c, 0), *(self.adj.len(x) for x in cheader)
+            )
+            fmt_values = self._format_col(i)
+            fmt_values = _make_fixed_width(
+                fmt_values, self.justify, minimum=header_colwidth, adj=self.adj
+            )
+
+            max_len = max(max(self.adj.len(x) for x in fmt_values), header_colwidth)
+            cheader = self.adj.justify(cheader, max_len, mode=self.justify)
+            strcols.append(cheader + fmt_values)
+
+        return strcols
 
-                max_len = max(max(self.adj.len(x) for x in fmt_values), header_colwidth)
-                cheader = self.adj.justify(cheader, max_len, mode=self.justify)
-                stringified.append(cheader + fmt_values)
+    def _get_strcols(self) -> List[List[str]]:
+        strcols = self._get_strcols_without_index()
 
-        strcols = stringified
+        str_index = self._get_formatted_index(self.tr_frame)
         if self.index:
             strcols.insert(0, str_index)
 
-        # Add ... to signal truncated
-        truncate_h = self.truncate_h
-        truncate_v = self.truncate_v
+        return strcols
 
-        if truncate_h:
-            col_num = self.tr_col_num
-            strcols.insert(self.tr_col_num + 1, [" ..."] * (len(str_index)))
-        if truncate_v:
-            n_header_rows = len(str_index) - len(frame)
-            row_num = self.tr_row_num
-            # cast here since if truncate_v is True, self.tr_row_num is not None
-            row_num = cast(int, row_num)
-            for ix, col in enumerate(strcols):
-                # infer from above row
-                cwidth = self.adj.len(strcols[ix][row_num])
+    def _to_str_columns(self) -> List[List[str]]:
+        """
+        Render a DataFrame to a list of columns (as lists of strings).
+        """
+        strcols = self._get_strcols()
+
+        if self.is_truncated:
+            strcols = self._insert_dot_separators(strcols)
+
+        return strcols
+
+    def _insert_dot_separators(self, strcols: List[List[str]]) -> List[List[str]]:
```

Comment:
you could define List[List[str]] at the top as an alias, maybe Screen, see if that is more clear

Reply:
woudln't rename strcols -> screen if you can do that

Reply:
I think that it is more clear that strcols is a a list of columns. If we rename it to screen, then that would not be clear. I think such rename may be more useful for the list of rows instead.